Content.Server/Medical/CryoPodSystem.cs

```diff
@@ -0,0 +1,367 @@
+
+        if (EntityManager.TryGetComponent(args.Used, out ToolComponent? tool)
+            && tool.Qualities.Contains("Prying")) // Why aren't those enums?
+        {
+            if (cryoPodComponent.IsPrying)
+                return;
+            cryoPodComponent.IsPrying = true;
+
+            _toolSystem.UseTool(args.Used, args.User, uid, 0f,
+                cryoPodComponent.PryDelay, "Prying",
+                new CryoPodPryFinished(uid, cryoPodComponent), new CryoPodPryInterrupted(uid, cryoPodComponent));
+
+            args.Handled = true;
+        }
+    }
+
+    private void OnCryoPodPryFinished(CryoPodPryFinished ev)
+    {
+        ev.CryoPodComponent.IsPrying = false;
+        EjectBody(ev.Uid, ev.CryoPodComponent);
+    }
+
+    private void OnCryoPodPryInterrupted(CryoPodPryInterrupted ev)
+    {
+        ev.CryoPodComponent.IsPrying = false;
+    }
+
+    private void AddAlternativeVerbs(EntityUid uid, CryoPodComponent cryoPodComponent, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract)
+            return;
+
+        // Eject verb
+        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+        {
+            args.Verbs.Add(new AlternativeVerb
+            {
+                Text = Loc.GetString("cryo-pod-verb-noun-occupant"),
+                Category = VerbCategory.Eject,
+                Priority = 1, // Promote to top to make ejecting the ALT-click action
+                Act = () => TryEjectBody(uid, args.User, cryoPodComponent)
+            });
+        }
+    }
+
+    private void OnExamined(EntityUid uid, CryoPodComponent component, ExaminedEvent args)
+    {
+        var container = _itemSlotsSystem.GetItem(component.Owner, "beakerSlot");
+        if (args.IsInDetailsRange && container != null && _solutionContainerSystem.TryGetFitsInDispenser(container.Value, out var containerSolution))
+        {
+            args.PushMarkup(Loc.GetString("cryo-pod-examine", ("beaker", Name(container.Value))));
+            if (containerSolution.CurrentVolume == 0)
+            {
+                args.PushMarkup(Loc.GetString("cryo-pod-empty-beaker"));
+            }
+        }
+    }
+
+    private void OnDestroyed(EntityUid uid, CryoPodComponent? cryoPodComponent, DestructionEventArgs args)
+    {
+        if (!Resolve(uid, ref cryoPodComponent))
+        {
+            return;
+        }
+        EjectBody(uid, cryoPodComponent);
+    }
+
+    private void OnEmagged(EntityUid uid, CryoPodComponent? cryoPodComponent, GotEmaggedEvent args)
+    {
+        if (!Resolve(uid, ref cryoPodComponent))
+        {
+            return;
+        }
+
+        cryoPodComponent.PermaLocked = true;
+        cryoPodComponent.Locked = true;
+        args.Handled = true;
+    }
+
+    private void OnPowerChanged(EntityUid uid, CryoPodComponent component, PowerChangedEvent args)
+    {
+        if (args.Powered)
+        {
+            EnsureComp<ActiveCryoPodComponent>(uid);
```

Comment:
Is this the proper way to add an Active component to an entity? In my initial tests, I used AddComp instead of EnsureComp. I tried to delete an entity from the debug verb and I had a weird runtime, "Attempted to add a ActiveCryoPodComponent component to an entity (cryopod) while it is terminating" (from AddComponentInternal). I didn't manage to replicate it, but I didn't see any special check being done elsewhere to prevent this case, or maybe I failed to find one.